### ISSUE TITLE
How to get sent_sg_parse?

### ISSUE THREAD

**Issue by cici-ai-club** [entalent/MemCap#3]
Hi, wentian, I am very interested in MemCap. I would like to get the sent_sg_parse since use_sg is True for the train. But I failed to find where and how to generate based on the code. I am thinking it is the part called "Stylized Sentence Decomposing" in the paper. Any implementation on how to generate that?

**Comment by Doragd** [comment]
I'm also waiting for wentian's help~do you successfully to preprocess the senticap datasets? I cannot find any code about the part.

**Comment by cici-ai-club** [comment]
> I'm also waiting for wentian's help~do you successfully to preprocess the senticap datasets? I cannot find any code about the part.

Not really. That part is not very clear yet. But I do see the Vocab and preprocess code files. It might help for that.  It will be awesome if we can get the preprocess from wentian. Temporarily cannot have a successful run through because of lacking files.  Since decomposing is essential part of the paper, I actually would like to play with it.

**Comment by n9705** [comment]
Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.

**Comment by Doragd** [comment]
> Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.

i think the code is truely written by the author of the paper !
you can see config.py line 19
`kenlm_build_path = '/home/wentian/library/kenlm/build'
`
wentian is the author of the paper.

**Comment by cici-ai-club** [comment]
> > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> 
> i think the code is truely written by the author of the paper !
> you can see config.py line 19
> `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> wentian is the author of the paper.

Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.

**Comment by Doragd** [comment]
> > > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> > 
> > 
> > i think the code is truely written by the author of the paper !
> > you can see config.py line 19
> > `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> > wentian is the author of the paper.
> 
> Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.

Though something provided, i cannot run the code simply. 😢 How about you? can you reproduce the result successfully？

**Comment by cici-ai-club** [comment]
> > > > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> > > 
> > > 
> > > i think the code is truely written by the author of the paper !
> > > you can see config.py line 19
> > > `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> > > wentian is the author of the paper.
> > 
> > 
> > Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.
> 
> Though something provided, i cannot run the code simply. 😢 How about you? can you reproduce the result successfully？

I can train the style classifier and language model he provided. I think that is very helpful for me. Hope you could get something too.

**Comment by Doragd** [comment]
> > > > > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> > > > 
> > > > 
> > > > i think the code is truely written by the author of the paper !
> > > > you can see config.py line 19
> > > > `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> > > > wentian is the author of the paper.
> > > 
> > > 
> > > Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.
> > 
> > 
> > Though something provided, i cannot run the code simply. 😢 How about you? can you reproduce the result successfully？
> 
> I can train the style classifier and language model he provided. I think that is very helpful for me. Hope you could get something too.

thanks your quick reply so much! i'm sorry for my reply cause i don't study the code carefully. i will study the code carefully.😄. Thank you and @entalent again!

**Comment by qin79** [comment]
> > > > > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> > > > 
> > > > 
> > > > i think the code is truely written by the author of the paper !
> > > > you can see config.py line 19
> > > > `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> > > > wentian is the author of the paper.
> > > 
> > > 
> > > Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.
> > 
> > 
> > Though something provided, i cannot run the code simply. 😢 How about you? can you reproduce the result successfully？
> 
> I can train the style classifier and language model he provided. I think that is very helpful for me. Hope you could get something too.

would you please give some guidance on running the code？thanks a lot！

**Comment by cici-ai-club** [comment]
I hope the author could get you back cause I am not the author of this paper and I only get hints of how they perform style evaluation in the styled_eval directory.

**Comment by cici-ai-club** [comment]
> > > > > > Is this code written by the author of the paper？The lack of README make me comfuse about the authority of this code.
> > > > > 
> > > > > 
> > > > > i think the code is truely written by the author of the paper !
> > > > > you can see config.py line 19
> > > > > `kenlm_build_path = '/home/wentian/library/kenlm/build' `
> > > > > wentian is the author of the paper.
> > > > 
> > > > 
> > > > Yeah, I agree, at least wentian provide something rather than nothing. Also, I hope more details if he or she had time.
> > > 
> > > 
> > > Though something provided, i cannot run the code simply. 😢 How about you? can you reproduce the result successfully？
> > 
> > 
> > I can train the style classifier and language model he provided. I think that is very helpful for me. Hope you could get something too.
> 
> would you please give some guidance on running the code？thanks a lot！

I hope the author could get you back cause I am not the author of this paper and I only get hints of how they perform style evaluation in the styled_eval directory.
